source.go

```diff
@@ -229,177 +207,96 @@ func (s *Source) populateIDs(ctx context.Context) error {
 			Msg("sleeping before checking for changes")
 		time.Sleep(s.config.pollInterval)
```

Comment:
Sidenote: another thing that doesn't have to be addressed in this PR - we should really use something like [`x/time/rate`](https://pkg.go.dev/golang.org/x/time/rate) for polling intervals, that goes for all connectors. Here the poll interval will be `s.config.pollInterval` + however long it took to create records in the previous batch. Additionally we can't guarantee that `Read` will be called continuously, a slow pipeline can create backpressure and delays between calls.

Reply:
Makes sense to me :+1: . I've created an issue for this: #40.